File: src/main/java/com/mercadez/service/AfiliadoService.java
package com.mercadez.service;

import com.mercadez.dto.*;
import com.mercadez.exception.*;
import com.mercadez.model.Afiliado;
import com.mercadez.repository.AfiliadoRepository;
import com.mercadez.security.JwtService;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
public class AfiliadoService {

    private final AfiliadoRepository repo;
    private final PasswordEncoder    encoder;
    private final JwtService         jwt;

    public AfiliadoService(AfiliadoRepository repo, PasswordEncoder encoder, JwtService jwt) {
        this.repo    = repo;
        this.encoder = encoder;
        this.jwt     = jwt;
    }

    @Transactional
    public AfiliadoResponse cadastrar(CadastroAfiliadoRequest req) {
        if (repo.existsByEmail(req.getEmail())) {
            throw new NegocioException("Email já cadastrado.");
        }
        if (repo.existsByCnpj(req.getCnpj())) {
            throw new NegocioException("CNPJ já cadastrado.");
        }

        Afiliado a = Afiliado.builder()
                .nomeProprietario(req.getNome_proprietario())
                .email(req.getEmail().toLowerCase())
                .senha(encoder.encode(req.getSenha()))
                .cnpj(req.getCnpj())
                .endereco(req.getEndereco())
                .telefone(req.getTelefone())
                .mercado(req.getMercado())
                .categoria(req.getCategoria())
                .funcionarios(req.getFuncionarios())
                .pagamento(req.getPagamento())
                .build();

        return AfiliadoResponse.de(repo.save(a));
    }

    @Transactional(readOnly = true)
    public LoginResponse login(LoginRequest req) {
        Afiliado a = repo.findByEmail(req.getEmail().toLowerCase())
                .orElseThrow(CredenciaisInvalidasException::new);

        if (!encoder.matches(req.getSenha(), a.getSenha())) {
            throw new CredenciaisInvalidasException();
        }

        String token = jwt.gerarToken(a.getId(), a.getEmail(), "AFILIADO");
        return new LoginResponse(token, "Bearer", "AFILIADO", a.getId(), a.getNomeProprietario(), a.getEmail());
    }

    @Transactional(readOnly = true)
    public AfiliadoResponse buscarPorId(Long id) {
        return AfiliadoResponse.de(
                repo.findById(id).orElseThrow(() -> new NaoEncontradoException("Afiliado não encontrado."))
        );
    }
}

File: app/services/afiliado_service.py
"""Equivalente Python de service/AfiliadoService.java"""
from sqlalchemy import select
from sqlalchemy.orm import Session

from app.core import security
from app.exceptions import CredenciaisInvalidasException, NaoEncontradoException, NegocioException
from app.models.afiliado import Afiliado
from app.schemas.afiliado import AfiliadoResponse, CadastroAfiliadoRequest
from app.schemas.usuario import LoginRequest, LoginResponse


def cadastrar(db: Session, req: CadastroAfiliadoRequest) -> AfiliadoResponse:
    email = req.email.lower()

    if db.scalar(select(Afiliado).where(Afiliado.email == email)) is not None:
        raise NegocioException("Email ja cadastrado.")

    if db.scalar(select(Afiliado).where(Afiliado.cnpj == req.cnpj)) is not None:
        raise NegocioException("CNPJ ja cadastrado.")

    afiliado = Afiliado(
        nome_proprietario=req.nome_proprietario,
        email=email,
        senha=security.hash_senha(req.senha),
        cnpj=req.cnpj,
        endereco=req.endereco,
        telefone=req.telefone,
        mercado=req.mercado,
        categoria=req.categoria,
        funcionarios=req.funcionarios,
        pagamento=req.pagamento,
    )
    db.add(afiliado)
    db.commit()
    db.refresh(afiliado)
    return AfiliadoResponse.de(afiliado)


def login(db: Session, req: LoginRequest) -> LoginResponse:
    email = req.email.lower()
    afiliado = db.scalar(select(Afiliado).where(Afiliado.email == email))

    if afiliado is None or not security.verificar_senha(req.senha, afiliado.senha):
        raise CredenciaisInvalidasException()

    token = security.gerar_token(afiliado.id, afiliado.email, "AFILIADO")
    return LoginResponse(
        token=token,
        tipo="Bearer",
        perfil="AFILIADO",
        id=afiliado.id,
        nome=afiliado.nome_proprietario,
        email=afiliado.email,
    )


def buscar_por_id(db: Session, afiliado_id: int) -> AfiliadoResponse:
    afiliado = db.get(Afiliado, afiliado_id)
    if afiliado is None:
        raise NaoEncontradoException("Afiliado nao encontrado.")
    return AfiliadoResponse.de(afiliado)
